master/internal/experiment_job_service_test.go

```diff
@@ -0,0 +1,38 @@
+package internal
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+)
+
+var rp = "mock"
+
+func TestResourcePool(t *testing.T) {
+	e := mockInternalExp()
+	require.Equal(t, mutexLocked(&e.mu), false)
+
+	rpName := e.ResourcePool()
+	require.Equal(t, rp, rpName)
+	require.Equal(t, mutexLocked(&e.mu), false)
+}
+
+func mutexLocked(m *sync.Mutex) bool {
+	state := reflect.ValueOf(m).Elem().FieldByName("state")
+	return state.Int()&1 == 1
+}
```

Comment:
I think by getting the ResourcePool you know the mutex is unlocked.

I think we should not go this deep on testing the mutex and just assuming if ResourcePool returns the right awsner the code is right